src/text/parsers/top_level.rs

```diff
@@ -40,9 +62,10 @@ pub(crate) fn version_int(input: &str) -> IonParseResult<&str> {
 /// https://amzn.github.io/ion-docs/docs/symbols.html#ion-version-markers
 pub(crate) fn ion_version_marker(input: &str) -> IonParseResult<(u32, u32)> {
     // See if the input text matches an IVM. If not, return a non-fatal error.
-    let (remaining_input, (_, major_text, _, minor_text)) = preceded(
+    let (remaining_input, (_, major_text, _, minor_text)) = delimited(
         whitespace_or_comments,
         tuple((tag("$ion_"), version_int, tag("_"), version_int)),
+        not(annotation_delimiter),
```

Comment:
If an IVM is followed by a `::`, it's an annotation and not an IVM. (e.g. `$ion_1_0::5`)